### PR TITLE
#2 Feature/input java files function

### PR DESCRIPTION
**Pull request by dvanderleij**
Add utility method for when you have a collection of java files and you do not want to analyse the whole repo.

### CODE REVIEW

**Review by mauricioaniche — CHANGES_REQUESTED** [review]
This PR contains the code of the other PR too. Can you make it clean (i.e., only the change) so that I can proper review it?

Maybe best is to close this PR and open another one.

**Comment by dvanderleij** [comment]
> This PR contains the code of the other PR too. Can you make it clean (i.e., only the change) so that I can proper review it?
> 
> Maybe best is to close this PR and open another one.

Yes, as soon as the other is merged i will make those commits dissapear from this PR, I would have put it in draft but it seems to be a feature only for organisations :)

File: src/main/java/com/github/mauricioaniche/ck/CK.java [hunk 1]
```diff
@@ -51,11 +50,18 @@ public CK() {
...
-		String[] srcDirs = FileUtils.getAllDirs(path);
 		String[] javaFiles = FileUtils.getAllJavaFiles(path);
-		String[] allDependencies = useJars ? FileUtils.getAllJars(path) : null;
-
 		log.info("Found " + javaFiles.length + " java files");
+
+		calculate(path, notifier, javaFiles);
+	}
+
+	public void calculate(String environment, CKNotifier notifier, String... javaFiles) {
```
Comment (mauricioaniche) [inline src/main/java/com/github/mauricioaniche/ck/CK.java:59]: Why is the variable called `environment`? Maybe use the same name in both `calculate` methods? One is called `path`. Just pick the one you prefer!

File: src/test/java/com/github/mauricioaniche/ck/integration/IntegrationTest.java [hunk 2]
```diff
@@ -54,7 +54,40 @@ public void notifyError(String sourceFilePath, Exception e) {
         });
     }
 
+    @Test
+    void testIndividualFiles() throws GitAPIException {
+        clone("https://github.com/apache/commons-lang.git");
```
Comment (mauricioaniche) [inline src/test/java/com/github/mauricioaniche/ck/integration/IntegrationTest.java:59]: Can this test run in one of the fixtures we have? This way we avoid cloning a very large project, and things would go faster!

**Review by mauricioaniche — CHANGES_REQUESTED** [review]
Great!

Some suggestions in the review!

File: src/test/java/com/github/mauricioaniche/ck/integration/IntegrationTest.java (hunk 2 above)
Comment (dvanderleij) [reply src/test/java/com/github/mauricioaniche/ck/integration/IntegrationTest.java]: I have changed the structure somewhat such that the tests are still separate so you see separate failures but the clone happens only once. Also I made it so the ck code does not need to be written twice.

File: src/main/java/com/github/mauricioaniche/ck/CK.java (hunk 1 above)
Comment (dvanderleij) [reply src/main/java/com/github/mauricioaniche/ck/CK.java]: I changed it to path :)

**Review by mauricioaniche — APPROVED** [review]
LGTM!

**Pull request merged**